janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java

```diff
@@ -186,9 +192,8 @@ public void deleteIndex(String indexName) throws IOException {
     public void bulkRequest(List<ElasticSearchMutation> requests) throws IOException {
         final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
         for (final ElasticSearchMutation request : requests) {
-            Map actionData = ImmutableMap.of(request.getRequestType().name().toLowerCase(),
-                ImmutableMap.of("_index", request.getIndex(), "_type", request.getType(), "_id", request.getId()));
-            outputStream.write(mapWriter.writeValueAsBytes(actionData));
+            final StringBuilder actionData = new StringBuilder("{\"").append(request.getRequestType().name().toLowerCase()).append("\":{\"_index\":\"").append(request.getIndex()).append("\", \"_type\" : \"").append(request.getType()).append("\", \"_id\" : \"").append(request.getId()).append("\"}}");
```

Comment:
are you building JSON by hand? can you use a library to do this to reduce the potential of syntax errors etc?